components/connector/internal/revocation/loader.go

```diff
@@ -0,0 +1,109 @@
+package revocation
+
+import (
+	"context"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+type Loader interface {
+	Run(ctx context.Context)
+}
+
+type revocationListLoader struct {
+	revocationListCache Cache
+	manager             Manager
+	configMapName       string
+	reconnectInternal   time.Duration
+}
+
+func NewRevocationListLoader(revocationListCache Cache,
+	manager Manager,
+	configMapName string,
+	reconnectInternal time.Duration,
+) Loader {
+	return &revocationListLoader{
+		revocationListCache: revocationListCache,
+		manager:             manager,
+		configMapName:       configMapName,
+		reconnectInternal:   reconnectInternal,
+	}
+}
+
+func (rl *revocationListLoader) Run(ctx context.Context) {
+	rl.startKubeWatch(ctx)
+}
+
+func (rl *revocationListLoader) startKubeWatch(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			log.Println("context cancelled, stop revocation config map watcher")
+			return
+		default:
```

Comment:
Instead of an empty default case, can we place the logic below in the default case itself? Maybe it'll look prettier. 😆

Reply:
It is standard to see empty default cases in situations like that. Also it is more readable for the happy path (scenario) not to be indented.

Reply:
Yes, it makes sense.